lua/entities/gmod_wire_expression2/init.lua

```diff
@@ -429,7 +429,7 @@ function ENT:Setup(buffer, includes, restore, forcecompile, filepath)
 		self:PCallHook('destruct')
 	end
 
-	self.uid = IsValid(self.player) and self.player:UniqueID() or "World"
+	self.uid = IsValid(self.player) and self.player:SteamID() or "World"
```

Comment:
shouldn't this be `AccountID` like the others?

Reply:
There's a reason why I use SteamID, see `PlayerAuth` hook:
https://github.com/wiremod/wire/pull/1904/commits/76ed3ddd68e51c918a4f20e078066ca40d426e74#diff-8b81bd658ab4c6beb2b84654d16f5fa4R563

It would require a conversion function to use AccountID for `.uid` field (as why I said it is an exception).

Reply:
I can add a conversion function to wire (there isn't a builtin one). And then use `AccountID` for `.uid` field, if you really want it.

Reply:
Why? `PlayerAuthed` gives a player object, unless `AccountID` doesn't return anything at that point?

Reply:
Please refer to
https://wiki.garrysmod.com/page/GM/PlayerAuthed
It is all fine here.

Reply:
That didn't answer my question, what's the point in using `SteamID` if you can get `AccountID` inside of `PlayerAuthed`?

Reply:
Oh yeah, you're right.
But, just using the 2nd argument (SteamID) seemed like a smaller change to me.
Let's not forget about integer overflow (`AccountID` is unsigned 32-bit integer) *cough*
It is probably not going to be of any concern any time soon, since it is not half way there yet, but at some point (decades later)...
I can switch it to `AccountID` without any problems tho, both works, what is it going to be?

Reply:
`AccountID` returns no value for singleplayer and bots too. Usage of `SteamID` is just simply the right thing for this part of code.

Reply:
> AccountID returns no value for singleplayer and bots too. Usage of SteamID is just simply the right thing for this part of code.

`SteamID` returns `"BOT"` for bots and `"NULL"` for singleplayer, so other than its ability to distinguish bots from singleplayer players (which we've never relied on anywhere) it's not any better. If you want to distinguish individual bots from each other, `SteamID64` is what you should use. `IsValid(self.player) and (self.player:AccountID() or 1) or 0` would work just as well as `IsValid(self.player) and self.player:SteamID() or "World"`.

> Let's not forget about integer overflow (AccountID is unsigned 32-bit integer) cough

This is irrelevant FUD - if there's a problem with it then we shouldn't use it anywhere. And all numbers in Garry's Mod Lua are 64-bit floating-point numbers, which can represent all integers up at least 2⁵².

Reply:
Are we going to treat bots as separate players in wire?

> This is irrelevant FUD - if there's a problem with it then we shouldn't use it anywhere. And all numbers in Garry's Mod Lua are 64-bit floating-point numbers, which can represent all integers up at least 2⁵².

Don't think just because a function returns a number, that a number is a double-floating point number!
`AccountID` **is** `uint32` behind the scenes. `uint32` is cast into `double` for Lua, so the integer overflow is thus possible. `uint32` number being cast into a `double` doesn't remove overflow issue (which will happen for it, some decades later).
Lua has no problems representing `uint32` as `double` number, but that's nothing to do with the fact of overflow.